### PR TITLE
Small optimalizations with count and strlen

### PR DESCRIPTION
**Pull request by snapshotpl**
Something more to @samsonasik pr in #125

### CODE REVIEW

File: src/Header/ContentType.php [hunk 1]
```diff
@@ -54,7 +54,7 @@ public static function fromString($headerLine)
         $mediaType         = array_shift($parts);
         $header = new static($value, trim($mediaType));
 
-        if (count($parts) > 0) {
+        if (!empty($parts)) {
```
Comment (samsonasik) [inline src/Header/ContentType.php]: space after not operator

File: src/Header/AbstractAccept.php [hunk 2]
```diff
@@ -170,7 +170,7 @@ protected function getParametersFromFieldValuePart($fieldValuePart)
             foreach ($paramsStrings as $param) {
                 $explode = explode('=', $param, 2);
 
-                if (count($explode) === 2) {
+                if (isset($explode[1])) {
```
Comment (Ocramius) [inline src/Header/AbstractAccept.php]: Should check that key 0 exists, and key 2 doesn't.

File: src/Header/AbstractAccept.php [hunk 3]
```diff
@@ -438,10 +438,12 @@ protected function sortFieldValueParts()
             //@todo count number of dots in case of type==application in subtype
 
             // So far they're still the same. Longest string length may be more specific
-            if (strlen($a->raw) == strlen($b->raw)) {
+            $aLength = strlen($a->raw);
+            $bLength = strlen($b->raw);
+            if ($aLength === $bLength) {
```
Comment (Ocramius) [inline src/Header/AbstractAccept.php]: Spaceship operator: only case where I ever suggested its usage.

File: src/Header/ContentType.php [hunk 4]
```diff
@@ -54,7 +54,7 @@ public static function fromString($headerLine)
         $mediaType         = array_shift($parts);
         $header = new static($value, trim($mediaType));
 
-        if (count($parts) > 0) {
+        if (! empty($parts)) {
             $parameters = [];
```
Comment (Ocramius) [inline src/Header/ContentType.php]: `empty()` not needed: the `! ` is sufficient

File: src/Header/Cookie.php [hunk 5]
```diff
@@ -60,7 +60,7 @@ public static function fromString($headerLine)
         $arrayInfo = [];
         foreach ($nvPairs as $nvPair) {
             $parts = explode('=', $nvPair, 2);
-            if (count($parts) != 2) {
+            if (! isset($parts[1])) {
```
Comment (Ocramius) [inline src/Header/Cookie.php]: Should check for key 2. Test needed

Comment (Ocramius) [reply src/Header/Cookie.php]: Never mind, explode is limited to 2 elements

File: src/Headers.php [hunk 6]
```diff
@@ -153,12 +153,22 @@ public function addHeaders($headers)
...
                 if (is_string($value)) {
                     $this->addHeaderLine($value);
-                } elseif (is_array($value) && count($value) == 1) {
-                    $this->addHeaderLine(key($value), current($value));
-                } elseif (is_array($value) && count($value) == 2) {
-                    $this->addHeaderLine($value[0], $value[1]);
-                } elseif ($value instanceof Header\HeaderInterface) {
+                    continue;
+                }
+                if (is_array($value)) {
```
Comment (Ocramius) [inline src/Headers.php]: Move contents of this block to a private method

File: src/PhpEnvironment/Request.php [hunk 7]
```diff
@@ -95,7 +95,7 @@ public function getContent()
     {
         if (empty($this->content)) {
             $requestBody = file_get_contents('php://input');
-            if (strlen($requestBody) > 0) {
+            if (! empty($requestBody)) {
```
Comment (Ocramius) [inline src/PhpEnvironment/Request.php]: `empty()` not needed: `!` is sufficient

Comment (samsonasik) [reply src/PhpEnvironment/Request.php]: for `bool` comparison usage, with `if ($requestBody) {` should be enough

File: src/Header/AbstractAccept.php (hunk 3 above)
Comment (weierophinney) [reply src/Header/AbstractAccept.php]: Can't use it here; this component still supports PHP 5.6.

File: src/Header/AbstractAccept.php (hunk 2 above)
Comment (weierophinney) [reply src/Header/AbstractAccept.php]: The explode operation was given a limit of 2, so there's no chance a key 2 will ever exist.

**Comment by weierophinney** [comment]
Closing, as the author has not incorporated any of the requested changes.

**Pull request closed without merging**